test/parse_test.js

```diff
@@ -52,331 +54,432 @@ describe('parse()', function() {
-      await expect(e.title).to.equal('Version 1.2.0 is not supported.');
-      await expect(e.detail).to.equal('Please use latest version of the specification.');
-      await expect(e.parsedJSON).to.deep.equal({ asyncapi: '1.2.0' });
-      await expect(e.validationErrors).to.deep.equal([{
-        jsonPointer: '/asyncapi',
-        startLine: 1,
-        startColumn: 1,
-        startOffset: 0,
-        endLine: 1,
-        endColumn: 16,
-        endOffset: offset(15, 1),
-      }]);
-    }
+    }, expectedErrorObject);
   });
 
   it('should fail when asyncapi is not yaml nor json', async function() {
-    try {
+    const expectedErrorObject = {
+      type: 'https://github.com/asyncapi/parser-js/invalid-yaml',
+      title: 'The provided YAML is not valid.',
+      detail: 'duplicated mapping key at line 2, column -4:\n    bad:\n    ^',
+      location: { startOffset: 5, startLine: 2, startColumn: -4 }
+    };
+
+    await checkErrorWrapper(async () => {
       await parser.parse('bad:\nbad:');
-    } catch (e) {
-      await expect(e.type).to.equal('https://github.com/asyncapi/parser-js/invalid-yaml');
-      await expect(e.title).to.equal('The provided YAML is not valid.');
-      await expect(e.detail).to.equal('duplicated mapping key at line 2, column -4:\n    bad:\n    ^');
-      await expect(e.location).to.deep.equal({ startOffset: 5, startLine: 2, startColumn: -4 });
-    }
+    }, expectedErrorObject);
   });
 
   it('should fail to resolve relative files when options.path is not provided', async function() {
-    const type = 'https://github.com/asyncapi/parser-js/dereference-error';
-    const message = `Error opening file "${path.resolve(process.cwd(), 'refs/refed.yaml')}" \nENOENT: no such file or directory, open '${path.resolve(process.cwd(), 'refs/refed.yaml')}'`;
-    const testFn = async () => { await parser.parse(inputYAML); };
-    await checkErrorTypeAndMessage(testFn, type, message);
-    await checkErrorParsedJSON(testFn, outputJsonWithRefs);
+    const expectedErrorObject = {
+      type: 'https://github.com/asyncapi/parser-js/dereference-error',
+      title: `Error opening file "${path.resolve(process.cwd(), 'refs/refed.yaml')}" \nENOENT: no such file or directory, open '${path.resolve(process.cwd(), 'refs/refed.yaml')}'`,
+      parsedJSON: outputJsonWithRefs,
+      refs: [
+        {
+          jsonPointer: '/components/schemas/testSchema/properties/test/$ref',
+          startLine: 35,
+          startColumn: 11,
+          startOffset: 736,
+          endLine: 35,
+          endColumn: 34,
+          endOffset: 759
+        }
+      ]
+    };
+    await checkErrorWrapper(async () => {
+      await parser.parse(inputYAML);
+    }, expectedErrorObject);
   });
 
   it('should offer information about YAML line and column where $ref errors are located', async function() {
-    try {
-      await parser.parse(inputYAML, { path: __filename });
```

Comment:
`{ path: __filename }` here and in other places dragged my attention when I noticed something is wrong with tests. Path was provided so the parser could not fail, the test was green even though in the `catch` clouse there were checks that should fail.